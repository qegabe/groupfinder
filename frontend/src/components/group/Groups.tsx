import React, { useEffect, useState } from "react";
import { Box, Button, Grid, Typography } from "@mui/material";
import GroupList from "./GroupList";
import SearchBar from "../common/SearchBar";
import { Link } from "react-router-dom";
import GroupFilter from "./GroupFilter";
import GroupFinderApi from "../../api";
import { Dayjs } from "dayjs";
import LoadingSpinner from "../common/LoadingSpinner";

interface FilterData {
  startTimeAfter: Dayjs | null;
  startTimeBefore: Dayjs | null;
  maxSize: number;
}

function Groups() {
  const [formData, setFormData] = useState<FilterData>({
    startTimeAfter: null,
    startTimeBefore: null,
    maxSize: 10,
  });
  const [groupsData, setGroupsData] = useState<ListGroup[]>([]);
  const [searchData, setSearchData] = useState<GroupFilter>({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    async function loadGroups() {
      setGroupsData(await GroupFinderApi.getGroups(searchData));
      setLoading(false);
    }
    loadGroups();
  }, [searchData]);

  function search(term: string) {
    const data: GroupFilter = {
      title: term,
      maxSize: formData.maxSize,
    };

    if (formData.startTimeAfter) {
      data.startTimeAfter = formData.startTimeAfter.toISOString();
    }
    if (formData.startTimeBefore) {
      data.startTimeBefore = formData.startTimeBefore.toISOString();
    }
    setSearchData(data);
    setLoading(true);
  }

  return (
    <Box sx={{ my: 4 }}>
      <Typography variant="h4">Search for groups</Typography>
      <Box component="div" sx={{ display: "flex", justifyContent: "center" }}>
        <SearchBar submitSearch={search} />
        <Button component={Link} to="/groups/new">
          Create
        </Button>
      </Box>

      <Grid container spacing={2}>
        <Grid item xs={8}>
          {loading ? <LoadingSpinner /> : <GroupList groupsData={groupsData} />}
        </Grid>
        <Grid item xs={4}>
          <GroupFilter formData={formData} setFormData={setFormData} />
        </Grid>
      </Grid>
    </Box>
  );
}

export default Groups;
