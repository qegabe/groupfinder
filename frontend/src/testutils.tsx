import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { RootState, setupStore } from "./store";
import { Provider } from "react-redux";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { LocalizationProvider } from "@mui/x-date-pickers";
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
import { PreloadedState } from "@reduxjs/toolkit";

interface CustomRenderOptions {
  currentRoute?: string;
  routePath?: string;
  preloadedState?: PreloadedState<RootState>;
}

function customRender(
  ui: ReactElement,
  cOptions: CustomRenderOptions = {},
  options?: Omit<RenderOptions, "wrapper">
) {
  const store = setupStore(cOptions.preloadedState);
  function AllTheProviders({ children }: { children: React.ReactNode }) {
    return (
      <Provider store={store}>
        <MemoryRouter
          initialEntries={
            cOptions.currentRoute ? [cOptions.currentRoute] : undefined
          }>
          <LocalizationProvider dateAdapter={AdapterDayjs}>
            {cOptions.routePath ? (
              <Routes>
                <Route path={cOptions.routePath} element={children} />
              </Routes>
            ) : (
              children
            )}
          </LocalizationProvider>
        </MemoryRouter>
      </Provider>
    );
  }
  return render(ui, { wrapper: AllTheProviders, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
