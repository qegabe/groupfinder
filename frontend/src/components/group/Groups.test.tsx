import React from "react";
import { render, waitFor, screen, fireEvent, within } from "../../testutils";
import "@testing-library/jest-dom";
import Groups from "./Groups";
import GroupFinderApi from "../../api";
import dayjs from "dayjs";

const mockGetGroups = jest
  .spyOn(GroupFinderApi, "getGroups")
  .mockImplementation(async () => {
    return [
      {
        id: 1,
        title: "Test Group 1",
        startTime: dayjs("2023-06-23T16:00:00.000Z"),
        endTime: dayjs("2023-06-23T17:00:00.000Z"),
      },
      {
        id: 2,
        title: "Test Group 2",
        startTime: dayjs("2023-06-23T17:00:00.000Z"),
        endTime: dayjs("2023-06-23T18:00:00.000Z"),
      },
      {
        id: 3,
        title: "Test Group 3",
        startTime: dayjs("2023-06-23T18:00:00.000Z"),
        endTime: dayjs("2023-06-23T19:00:00.000Z"),
      },
    ];
  });

const mockSearchGame = jest
  .spyOn(GroupFinderApi, "searchGame")
  .mockImplementation(async () => {
    return [{ id: 1, title: "Test Game", coverUrl: "" }];
  });

afterEach(() => {
  jest.clearAllMocks();
});

it("renders", async () => {
  const { container } = render(<Groups />);
  await waitFor(() => {
    expect(container.querySelector(".LoadingSpinner")).not.toBeInTheDocument();
  });
  expect(mockGetGroups).toHaveBeenCalledTimes(1);
  expect(screen.getByText("Test Group 1")).toBeInTheDocument();
});

it("matches snapshot", async () => {
  const { container, asFragment } = render(<Groups />);
  await waitFor(() => {
    expect(container.querySelector(".LoadingSpinner")).not.toBeInTheDocument();
  });
  expect(asFragment()).toMatchSnapshot();
});

it("lets users filter groups", async () => {
  const { container } = render(<Groups />);
  await waitFor(() => {
    expect(container.querySelector(".LoadingSpinner")).not.toBeInTheDocument();
  });
  const searchField = container.querySelector("#term");
  const maxMembers = screen.getByLabelText("Max Members");
  const searchButton = screen.getByText("Search");
  fireEvent.change(searchField as Element, { target: { value: "a" } });
  fireEvent.change(maxMembers, { target: { value: 15 } });
  fireEvent.click(searchButton);
  await waitFor(() => {
    expect(mockGetGroups).toBeCalledWith({ title: "a", maxSize: 15 });
  });
});

it("lets users add game to filter", async () => {
  const { container } = render(<Groups />);
  await waitFor(() => {
    expect(container.querySelector(".LoadingSpinner")).not.toBeInTheDocument();
  });

  const autocomplete = screen.getByTestId("autocomplete-addgame");
  const addGameForm = screen.getByTestId("form-addgame");
  const input = within(addGameForm).getByLabelText("Add a game");
  const add = within(addGameForm).getByText("Add");

  autocomplete.focus();
  fireEvent.change(input, { target: { value: "a" } });
  await waitFor(() => {
    const game = screen.getByText("Test Game");
    expect(mockSearchGame).toHaveBeenCalledTimes(1);
    expect(game).toBeInTheDocument();
    fireEvent.click(game);
  });
  fireEvent.click(add);
  await waitFor(() => {
    expect(screen.getByTestId("gamecard-1")).toBeInTheDocument();
  });

  const searchButton = screen.getByText("Search");
  fireEvent.click(searchButton);
  await waitFor(() => {
    expect(mockGetGroups).toBeCalledWith({
      title: "",
      maxSize: 10,
      hasGames: [1],
    });
  });
});
