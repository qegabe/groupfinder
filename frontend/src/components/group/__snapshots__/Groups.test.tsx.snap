// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-j26pjy"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1bnhfwg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-capmjd-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Search for groups
          </h4>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          href="/groups/new"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Create Group
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-tcrs79-MuiGrid-root"
      >
        <form
          autocomplete="off"
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-o4pm7k-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="term"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1k2rr11-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-tcrs79-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-controls="users-content"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
            id="users-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Filter
              </p>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="users-header"
                  class="MuiAccordion-region"
                  id="users-content"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                  >
                    <div
                      class="MuiStack-root css-1p5q5e5-MuiStack-root"
                    >
                      <form
                        class="MuiBox-root css-1etuyqg"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1ge1fqd-MuiButtonBase-root-MuiButton-root"
                          tabindex="0"
                          type="button"
                        >
                          Reset
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <div
                          class="MuiFormControl-root MuiTextField-root css-1akjbnu-MuiFormControl-root-MuiTextField-root"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                            data-shrink="false"
                            for=":r8:"
                            id=":r8:-label"
                          >
                            Starting Before
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                          >
                            <input
                              aria-invalid="false"
                              autocomplete="off"
                              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                              id=":r8:"
                              inputmode="text"
                              placeholder="MM/DD/YYYY hh:mm aa"
                              type="text"
                              value=""
                            />
                            <div
                              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                            >
                              <button
                                aria-label="Choose date"
                                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                                tabindex="0"
                                type="button"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                  data-testid="CalendarIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                                  />
                                </svg>
                                <span
                                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                />
                              </button>
                            </div>
                            <fieldset
                              aria-hidden="true"
                              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="css-yjsfm1"
                              >
                                <span>
                                  Starting Before
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                        <div
                          class="MuiFormControl-root MuiTextField-root css-1akjbnu-MuiFormControl-root-MuiTextField-root"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                            data-shrink="false"
                            for=":ra:"
                            id=":ra:-label"
                          >
                            Starting After
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                          >
                            <input
                              aria-invalid="false"
                              autocomplete="off"
                              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                              id=":ra:"
                              inputmode="text"
                              placeholder="MM/DD/YYYY hh:mm aa"
                              type="text"
                              value=""
                            />
                            <div
                              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                            >
                              <button
                                aria-label="Choose date"
                                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                                tabindex="0"
                                type="button"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                  data-testid="CalendarIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                                  />
                                </svg>
                                <span
                                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                />
                              </button>
                            </div>
                            <fieldset
                              aria-hidden="true"
                              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="css-yjsfm1"
                              >
                                <span>
                                  Starting After
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                        <div
                          class="MuiBox-root css-0"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
                          >
                            Max Members
                          </p>
                          <span
                            class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-zugq7j-MuiSlider-root"
                            id="maxSize"
                          >
                            <span
                              class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
                            />
                            <span
                              class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                              style="left: 0%; width: 18.367346938775512%;"
                            />
                            <span
                              class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
                              data-index="0"
                              style="left: 18.367346938775512%;"
                            >
                              <input
                                aria-label="Max Members"
                                aria-orientation="horizontal"
                                aria-valuemax="50"
                                aria-valuemin="1"
                                aria-valuenow="10"
                                data-index="0"
                                max="50"
                                min="1"
                                step="1"
                                style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                                type="range"
                                value="10"
                              />
                              <span
                                aria-hidden="true"
                                class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
                              >
                                <span
                                  class="MuiSlider-valueLabelCircle"
                                >
                                  <span
                                    class="MuiSlider-valueLabelLabel"
                                  >
                                    10
                                  </span>
                                </span>
                              </span>
                            </span>
                          </span>
                        </div>
                      </form>
                      <div
                        class="MuiBox-root css-xfk83p"
                      >
                        <div
                          class="MuiBox-root css-1l4w6pd"
                          data-testid="div-addgame"
                        >
                          <div
                            class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-52llsg-MuiAutocomplete-root"
                            data-testid="autocomplete-addgame"
                          >
                            <div
                              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                            >
                              <label
                                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                                data-shrink="false"
                                for=":rc:"
                                id=":rc:-label"
                              >
                                Add a game
                              </label>
                              <div
                                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                              >
                                <input
                                  aria-autocomplete="both"
                                  aria-expanded="false"
                                  aria-invalid="false"
                                  autocapitalize="none"
                                  autocomplete="off"
                                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                                  id=":rc:"
                                  role="combobox"
                                  spellcheck="false"
                                  type="text"
                                  value=""
                                />
                                <div
                                  class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                                >
                                  <button
                                    aria-label="Open"
                                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                                    tabindex="-1"
                                    title="Open"
                                    type="button"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                      data-testid="ArrowDropDownIcon"
                                      focusable="false"
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        d="M7 10l5 5 5-5z"
                                      />
                                    </svg>
                                    <span
                                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                    />
                                  </button>
                                </div>
                                <fieldset
                                  aria-hidden="true"
                                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                >
                                  <legend
                                    class="css-yjsfm1"
                                  >
                                    <span>
                                      Add a game
                                    </span>
                                  </legend>
                                </fieldset>
                              </div>
                            </div>
                          </div>
                          <button
                            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
                            tabindex="0"
                            type="button"
                          >
                            Add
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <div
                          class="MuiBox-root css-1ebprri"
                        >
                          <div
                            class="MuiStack-root css-76scyl-MuiStack-root"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-tcrs79-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Test Group 1
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Starting 10 days ago
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                href="/groups/1"
                tabindex="0"
              >
                Details
              </a>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Test Group 2
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Starting 10 days ago
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                href="/groups/2"
                tabindex="0"
              >
                Details
              </a>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Test Group 3
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Starting 10 days ago
              </p>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                href="/groups/3"
                tabindex="0"
              >
                Details
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
